### PR TITLE
Add DatePicker variant with no text field

### PR DESCRIPTION
**Pull request by amygl**
#493 DatePicker variant - No input text box
Added a DatePicker variant that will hide the text field, it will be opened via an external action (like a button click).

Added 3 optional props:
showDatePicker - When true the DatePicker's state will be updated to display the date picker.
hideTextField - When true text field will be hidden
onDismiss - callback for when the DatePicker is closed

### CODE REVIEW

**Comment by dzearing** [comment]
A general comment would be to split the component into smaller ones, instead of providing a "no input box" flag. That might be more elegant.

**Comment by amygl** [comment]
I've refactored the code so that there is a separate Calendar component that is used by the DatePicker. This component can then be hooked up to a button or text field or other control.

**Comment by dzearing** [comment]
Thanks Amy! I will deploy and play with it over the weekend when i get a chance.

**Comment by MLoughry** [comment]
It seems that focus isn't being placed in the date picker when it opens, which is a regression from current behavior.

**Comment by MLoughry** [comment]
You should add an example (or modify an existing one) to show off the `isMonthPickerVisible` option.

**Comment by MLoughry** [comment]
Unrelated to your change, so feel free to ignore: we're not using `Layer` or `Callout` or similar logic for the Calendar, so it's getting clipped in the example pages.

If you're willing to do that work, it'd be awesome!

**Comment by amygl** [comment]
I have fixed the focus regression and added an example to the Calendar demo page to show when the month picker is not visible.

**Comment by MLoughry** [comment]
The general practice has been to conditional setting focus on actions where we explicitly expect focus to be set. Otherwise, a change in the props passed by the parent causes the Calendar to rerender and steal focus, even if that's not desired (for example, changing the month using the arrows in the day picker steals focus before the user can scroll further through the months). 

In the calendar case, I think it should set focus on update in the case of 'go to today' or selecting a month in the month picker. On mount, there should be a property dictating whether to set focus (which can be true by default).

**Comment by amygl** [comment]
I've updated DatePicker to host Calendar in a callout, fixed up the focus issue, and added a calendar test.
The calendar test only tests rendering, I ran out of time to add a functionality test.

**Comment by dzearing** [comment]
Addresses https://github.com/OfficeDev/office-ui-fabric-react/issues/331

File: packages/office-ui-fabric-react/src/components/Calendar/Calendar.tsx [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+import './Calendar.scss';
+
+export interface ICalendarState {
+    /** The currently focused date in the calendar, but not necessarily selected */
+    navigatedDate?: Date;
+    /** The currently selected date in the calendar */
+    selectedDate?: Date;
+}
+
+export class Calendar extends BaseComponent<ICalendarProps, ICalendarState> {
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Calendar/Calendar.tsx]: nit: we use 2 spaces.

**Pull request merged**